pkg/riff/commands/application_status.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/projectriff/riff/pkg/cli"
+	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ApplicationStatusOptions struct {
+	cli.ResourceOptions
+}
+
+var (
+	_ cli.Validatable = (*ApplicationStatusOptions)(nil)
+	_ cli.Executable  = (*ApplicationStatusOptions)(nil)
+)
+
+func (opts *ApplicationStatusOptions) Validate(ctx context.Context) *cli.FieldError {
+	errs := cli.EmptyFieldError
+
+	errs = errs.Also(opts.ResourceOptions.Validate(ctx))
+
+	return errs
+}
+
+func (opts *ApplicationStatusOptions) Exec(ctx context.Context, c *cli.Config) error {
+	app, err := c.Build().Applications(opts.Namespace).Get(opts.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	now := time.Now()
+	ready := app.Status.GetCondition(buildv1alpha1.ApplicationConditionReady)
+	info := make([]cli.Info, 7)
+	info[0] = cli.Info{"Name", app.Name}
+	info[1] = cli.Info{"Namespace", app.Namespace}
+	info[2] = cli.Info{"Status", cli.FormatConditionStatus(app.Status.GetCondition(buildv1alpha1.ApplicationConditionReady))}
+	src := ""
+	if app.Spec.Source != nil && app.Spec.Source.Git != nil {
+		src = app.Spec.Source.Git.URL
+	}
+	info[3] = cli.Info{"Source", cli.FormatEmptyString(src)}
+	info[4] = cli.Info{"Target Image", cli.FormatEmptyString(app.Spec.Image)}
+	info[5] = cli.Info{"Latest Image", cli.FormatEmptyString(app.Status.LatestImage)}
+	info[6] = cli.Info{"Age", cli.FormatTimestampSince(app.CreationTimestamp, now)}
+	cli.PrintStatusMessage(c, "Application", info, false)
+	build := app.Status.GetCondition(buildv1alpha1.ApplicationConditionBuildSucceeded)
+	cond := ready
+	if build != nil && build.Status != v1.ConditionUnknown {
+		cond = build
+	}
+	if cond != nil {
+		cli.PrintStatusMessage(c, "Status Info", cli.StatusConditionInfo(cond), true)
```

Comment:
The name `Status Info` is problematic because the whole command is called status and prints info. The section header adds no meaningful value other than a visual break between the previous fields. The info it is printing is for the ready condition.